src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -36,7 +36,7 @@ internal sealed class TypeCache : MarshalByRefObject
     /// <summary>
     /// ClassInfo cache.
     /// </summary>
-    private readonly ConcurrentDictionary<string, TestClassInfo> _classInfoCache = new();
+    private readonly ConcurrentDictionary<string, TestClassInfo?> _classInfoCache = new();
```

Comment:
Why can the value be null now?

Reply:
Previously, we were not caching null values, but the method that returns the info had a code path that returns null (it was an un-cached code path)